_docs/developer/developing_the_php_site/modules.md

```diff
@@ -0,0 +1,22 @@
+---
+title: ES6 Modules and Typescript
+---
+
+[ES6 Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) allow JavaScript code to be split up
+into "modules" that can allow other modules to import functions and variables. Modules in Submitty are loaded the 
```

Comment:
I think it's important to note here that `ES6 Modules` do not have access to the global scope, something like:

> Unlike regular JS files, ES6 modules are not included into the global JS scope, and are self-contained. When a module is executed, it only has access to functions that it imports, and that the HTML cannot call functions defined within modules. As such, to hook a function defined in a module to something in the DOM, such as a click event listener, you will need to use the [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) API.